editor/editor_quick_open.cpp

```diff
@@ -75,7 +80,9 @@ void EditorQuickOpen::_build_search_cache(EditorFileSystemDirectory *p_efsd) {
 		// Iterate all possible base types.
 		for (String &parent_type : base_types) {
 			if (ClassDB::is_parent_class(engine_type, parent_type) || EditorNode::get_editor_data().script_class_is_parent(script_type, parent_type)) {
-				files.push_back(file.substr(6, file.length()));
+				if (!(skip_addons && file.substr(6).begins_with(addons_directory + "/"))) {
```

Comment:
```suggestion
				if (!skip_addons || !file.substr(6).begins_with(addons_directory + "/")) {
```
Simpler and easier to read than the grouped form IMO